### PR TITLE
Scamp connection cleanup.

### PR DESCRIPTION
**Pull request by Christian-B**
Part of https://github.com/SpiNNakerManchester/SpiNNMan/pull/266
See desciption there

### CODE REVIEW

File: spinn_front_end_common/interface/interface_functions/spalloc_allocator.py [hunk 1]
```diff
@@ -141,28 +142,43 @@ def spalloc_allocator(spalloc_server, n_chips=None, n_boards=None):
...
+                f"Asking for new job as {hostname} "
+                f"as in the spalloc_avoid_boards list")
+            job, hostname = _launch_job(n_boards, spalloc_kw_args)
+        else:
+            break
+    if avoid_boards:
+        for key in list(connections.keys()):
+            if connections[key] in avoid_boards:
+                logger.warning(
+                    f"Remvoing connection info for {connections[key]}")
```
Comment (rowleya) [inline spinn_front_end_common/interface/interface_functions/spalloc_allocator.py:177]:
```suggestion
                    f"Removing connection info for {connections[key]} as in the spalloc avoid_boards list")
```

File: spinn_front_end_common/interface/provenance/db.sql [hunk 2]
```diff
@@ -202,4 +202,10 @@ CREATE TABLE IF NOT EXISTS connector_provenance(
     description STRING NOT NULL,
     the_value INTEGER NOT NULL);
 
-
+---------------------------------------------------------------------
+-- A table to store job.info
+CREATE TABLE IF NOT EXISTS boards_provenanc(
```
Comment (rowleya) [inline spinn_front_end_common/interface/provenance/db.sql:207]:
```suggestion
CREATE TABLE IF NOT EXISTS boards_provenance(
```

File: spinn_front_end_common/interface/provenance/provenance_writer.py [hunk 3]
```diff
@@ -276,3 +276,21 @@ def insert_connector(
...
+
+        :param connections: {(x, y): hostname, ...} or None
+        :type connections: dict((int, int): str) or None
+        """
+        if not connections:
+            return
+        with self.transaction() as cursor:
+            cursor.executemany(
+                """
+                INSERT OR IGNORE INTO boards_provenanc(
```
Comment (rowleya) [inline spinn_front_end_common/interface/provenance/provenance_writer.py:292]:
```suggestion
                INSERT OR IGNORE INTO boards_provenance(
```

File: spinn_front_end_common/interface/spinnaker.cfg [hunk 4]
```diff
@@ -117,7 +117,8 @@ spalloc_port = 22244
 spalloc_user = None
 spalloc_machine = None
 # A comma seperated list of boards to avoid as chip 0,0
-spalloc_avoid_boards = 10.11.195.1
+# spalloc_avoid_boards = 10.11.195.1
```
Comment (rowleya) [inline spinn_front_end_common/interface/spinnaker.cfg:120]:
```suggestion
```

Comment (rowleya) [reply spinn_front_end_common/interface/spinnaker.cfg:120]: i.e. no need to keep this

**Review by rowleya — CHANGES_REQUESTED** [review]
Minor changes requested - all done as suggestions ;)

File: spinn_front_end_common/interface/spinnaker.cfg (hunk 4 above)
Comment (Christian-B) [reply spinn_front_end_common/interface/spinnaker.cfg:120]: Was kept as a format

**Comment by dkfellows** [comment]
Interacts with #842

**Pull request merged**